test/api_mocker.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Class to create a fake helper library http client request method
+ * @param {object[]} options
+ * @return {function}
+ */
+class ApiMocker {
```

Comment:
It seems to me we are reinventing the wheel here. Instead of this custom api mocker, lets use a proper mocking library like [sinon.js](https://sinonjs.org/).